### ISSUE TITLE
Binary Classification as final output

### ISSUE THREAD

**Issue by MC-Dave** [Nixtla/neuralforecast#385]
I have a use case wherein I need to take multi-variate time-series data as input, find correlations, and predict an output. In my particular case, I only need the output to be a 0 or 1. 0 meaning down, and 1 meaning up.

NeuralForecast models appear to be designed to forecast an unbounded value(s). Instead of having a fixed label applied to each observation in the dataset, the `horizon` parameter is used to determine the x-shift to produce the y labels at runtime. 

Is it possible to update the output so that it gives a value between 0 and 1, or each predicted time period? 

Thank you for your time! NeuralForecast is an excellent package

**Comment by kdgutier** [comment]
Hi @MC-Dave,

In the last weeks we added a new type of modular [DistributionLoss](https://github.com/Nixtla/neuralforecast/blob/main/nbs/losses.pytorch.ipynb).

Currently it only has Poisson, StudentT and Normal distributions available, but it might be simple to create a custom module for Bernoulli output, inheriting from the [Pytorch Distributions](https://pytorch.org/docs/stable/distributions.html).

**Comment by MC-Dave** [comment]
@kdgutier Thank you for the response and info. I will have to dig into it further. I am not sure how a loss unction affect the output of the model at the point, but am unfamiliar with the relevant implementations at present. 

Thank you for the references! I will take a look and see if I can get the output I'm looking for.

**Comment by MC-Dave** [comment]
@kdgutier Could I conscript you to help me in writing up a custom distribution loss module to implement Bernoulli output? 

I am less than familiar with the implementation specifics of distribution loss functions. 

I am happy to compensate you for your time. If interested you can reach me at -> dmccarty@protonmail.com

**Comment by kdgutier** [comment]
Hi @MC-Dave,

I am working in this branch: https://github.com/Nixtla/neuralforecast/tree/tweedie_loss
That has Zero Inflated and Bernoulli categorical distribution losses.

Additionally I added an example of a Bernoulli output in the end of this notebook:
`nbs/models.nhits.ipynb `

Hope it helps

**Comment by MC-Dave** [comment]
@kdgutier Thank you very much for all this work! It is most helpful. Working on getting it implemented into my local project. Thank you again!

**Comment by MC-Dave** [comment]
@kdgutier Thanks again. The Bernoulli DistributionLoss appears to be working. I am curious about your example in the [nhits.ipynb](https://github.com/Nixtla/neuralforecast/blob/tweedie_loss/nbs/models.nhits.ipynb) file.

You specify a `loss` and a `valid_loss` function. `DistributionLoss` and `Accuracy` respecitvely (when using Bernoulli distribution)

For the `NBEATSx` module there is only a `loss` function and no ability to specify a separate validation loss function. 

Should there be a separate `valid_loss` argument? 

My NBEATSx definition (truncated)
```
        NBEATSx(
            h=horizon,
            input_size=input_size,
            futr_exog_list=future_exog, 
            hist_exog_list=historic_exog,  

            learning_rate=1e-5,
            batch_size=1024,
            ...

            loss=DistributionLoss(distribution='Bernoulli', quantiles=[0, 1], return_params=True),
            scaler_type="robust",

            early_stop_patience_steps=4,
            val_check_steps=10,
        )
```

from nhits.ipynb

```
model = MLP(h=12,
            input_size=24,
            loss=DistributionLoss(distribution='Bernoulli', level=[80, 90], return_params=True),
            valid_loss=Accuracy(),
            stat_exog_list=['airline1'],
            scaler_type='robust',
            max_steps=200,
            early_stop_patience_steps=2,
            val_check_steps=10,
            learning_rate=1e-3)
```

**Comment by kdgutier** [comment]
In principle all the [neural forecast models collection](https://nixtla.github.io/neuralforecast/models.html), should be able to receive a training loss `loss`, and a validation `valid_loss` metric with which you can monitor the model's performance during its optimization.

Let us know if there are models that can't receive the validation loss.
Here is the `NBEATSx` documentation: https://nixtla.github.io/neuralforecast/models.nbeatsx.html

**Comment by MC-Dave** [comment]
@kdgutier After including Accuracy (from your working branch mentioned above) as the `valid_loss` of the `NBEATSx` module, I have a question regarding `y` and `y_hat` that I'm seeing when computing the Accuracy loss.

Within the classes `__call__` function, the `y` and `y_hat` being compared look like the following, respectively

y
```
       [[[1.],
         [0.],
         [0.],
         [0.],
         [1.],
         [1.]],
```

y_hat
```
       [[[0., 1.],
         [0., 1.],
         [0., 1.],
         [0., 1.],
         [0., 1.],
         [0., 1.]],
```

The shape mismatch doesn't cause any errors but it does seem odd to me. 

Let me know if this looks correct. Thanks again for all your help!

**Comment by kdgutier** [comment]
Thanks for pointing the difference in the one-hot vs categorical encoding of the target variable.
I will fix it.
